fendermint/vm/message/src/chain.rs

```diff
@@ -11,8 +12,7 @@ use crate::signed::SignedMessage;
 /// signed by BLS signatures are aggregated to the block level, and their original
 /// signatures are stripped from the messages, to save space. Tendermint Core will
 /// not do this for us (perhaps with ABCI++ Vote Extensions we could do it), though.
-#[derive(Clone, Debug, Serialize, Deserialize, Hash)]
-#[serde(untagged)]
```

Comment:
This has been removed, because if we leave it we get:
```
Error { description: "Msg(\"Only bytes can be deserialized into a CID\")", protocol: Cbor }
```
This was expected, and exactly why I wanted to add these types even though I won't use them yet. 

What I didn't expect was that I'd get this error even if there was only one variant wrapping a CID. I thought serde [will try one, then the other](https://serde.rs/enum-representations.html#untagged), and return whichever variant works. 

I am not sure what the tagging mechanism will look like when combined with IPLD. I can generate examples but can't write a "spec".